src/lib/apollo/sessionData.js

```diff
@@ -0,0 +1,75 @@
+import { Buffer } from "buffer";
+import Cookies from "js-cookie";
+
+/**
+ * @name getSessionData
+ * @method
+ * @private
+ * @return {Object} session data
+ */
+export function getUserSessionData() {
+  let sessionData = {};
+  const data = Cookies.get("storefront-session");
+  if (!data) return {};
+
+  try {
+    const expanded = Buffer.from(data, "base64").toString("utf8");
+    sessionData = JSON.parse(expanded, (key, value) => {
+      if (key === "user" && typeof value === "string") return JSON.parse(value);
+      return value;
+    });
+  } catch (error) {
+    // eslint-disable-next-line no-console
+    console.error(`Error while reading session data: ${error}`);
+  }
+  if (typeof sessionData.passport !== "object") {
+    return {};
+  }
+  return sessionData.passport.user;
+}
+
+/**
+ * @name getOAuthClientConfig
+ * @method
+ * @private
+ * @return {Object} client config data
+ */
+export function getOAuthClientConfig() {
+  let clientConfig = {};
+  const data = Cookies.get("oauth_client_config");
+  if (!data) return {};
+
+  try {
+    const expanded = Buffer.from(data, "base64").toString("utf8");
+    clientConfig = JSON.parse(expanded);
+  } catch (error) {
+    // eslint-disable-next-line no-console
+    console.error(`Error while reading session data: ${error}`);
+  }
+  return clientConfig;
+}
+
+/**
+ * @name createNewSessionData
+ * @method
+ * @private
+ * @param {Object} json session data
+ * @return {String} encoded string containing new session info
+ */
+export function createNewSessionData(json) {
+  let newSessionInfo = "";
+  try {
+    newSessionInfo = Buffer.from(JSON.stringify({
```

Comment:
I don't think most browsers have `Buffer`, but it looks like this is being called from browser code.

Reply:
Yes, for that I added a browser implementation of Buffer. https://github.com/feross/buffer.

Reply:
Oh ok I see the import now